ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -15,35 +15,43 @@ import Position.NoPosition
 import Types._
 
 object Trees {
-  /** AST node of the IR. */
-  abstract sealed class Tree {
-    val pos: Position
-    val tpe: Type
+  /** Base class for all nodes in the IR.
+   *
+   *  Usually, one of the direct subclasses of `IRNode` should be used instead.
+   */
+  abstract sealed class IRNode {
+    def pos: Position
 
     def show: String = {
       val writer = new java.io.StringWriter
       val printer = new Printers.IRTreePrinter(writer)
-      printer.print(this)
+      printer.printAnyNode(this)
       writer.toString()
     }
   }
 
+  /** Node for a statement or expression in the IR. */
+  abstract sealed class Tree extends IRNode {
+    val tpe: Type
+  }
+
   // Identifiers and properties
 
-  sealed trait PropertyName {
+  sealed trait PropertyName extends IRNode {
     /** Encoded name of this PropertyName within its owner's scope.
      *
      *  For [[ComputedName]]s, the value of `encodedName` cannot be relied on
      *  beyond equality tests, and the fact that it starts with `"__computed_"`.
      */
     def encodedName: String
-
-    def pos: Position
   }
 
   case class Ident(name: String, originalName: Option[String])(
-      implicit val pos: Position) extends PropertyName {
+      implicit val pos: Position)
+      extends IRNode with PropertyName {
```

Comment:
You mean because `PropertyName` already extends `IRNode` on its own? It feels more explicit to me to specify `extends IRNode` here nevertheless. Same for `ComputedName`.